spec/iso8601/time_interval_spec.rb

```diff
@@ -0,0 +1,375 @@
+# encoding: utf-8
+
+require 'spec_helper'
+
+RSpec.describe ISO8601::TimeInterval do
+  describe 'pattern initialization' do
+    it "should raise a ISO8601::Errors::UnknownPattern if it not a valid interval pattern" do
+      # Invalid separators
+      expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00ZP1Y2M10DT2H30M') }
+        .to raise_error(ISO8601::Errors::UnknownPattern)
+      expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z-P1Y2M10D') }
+        .to raise_error(ISO8601::Errors::UnknownPattern)
+      expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z~P1Y2M10D') }
+        .to raise_error(ISO8601::Errors::UnknownPattern)
+      expect { ISO8601::TimeInterval.new('P1Y2M10DT2H30M2007-03-01T13:00:00Z') }
+        .to raise_error(ISO8601::Errors::UnknownPattern)
+      expect { ISO8601::TimeInterval.new('P1Y2M10D-2007-03-01T13:00:00Z') }
+        .to raise_error(ISO8601::Errors::UnknownPattern)
+      expect { ISO8601::TimeInterval.new('P1Y2M10D~2007-03-01T13:00:00Z') }
+        .to raise_error(ISO8601::Errors::UnknownPattern)
+      expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z2008-05-11T15:30:00Z') }
+        .to raise_error(ISO8601::Errors::UnknownPattern)
+      expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z-2008-05-11T15:30:00Z') }
+        .to raise_error(ISO8601::Errors::UnknownPattern)
+      expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z~2008-05-11T15:30:00Z') }
+        .to raise_error(ISO8601::Errors::UnknownPattern)
+    end
+
+    describe 'with duration' do
+      it "should raise a ISO8601::Errors::UnknownPattern for any unknown pattern" do
+        expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/') }
+          .to raise_error(ISO8601::Errors::UnknownPattern)
+        expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/P') }
+          .to raise_error(ISO8601::Errors::UnknownPattern)
+        expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT') }
+          .to raise_error(ISO8601::Errors::UnknownPattern)
+      end
+    end
+
+    describe 'with DateTimes' do
+      it "should raise a ISO8601::Errors::UnknownPattern for any unknown pattern" do
+        expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/2010-0-09') }
+          .to raise_error(ISO8601::Errors::UnknownPattern)
+        expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/2010-05-09T103012+0400') }
+          .to raise_error(ISO8601::Errors::UnknownPattern)
+        expect { ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/2014-W15-02T10:11:12Z') }
+          .to raise_error(ISO8601::Errors::UnknownPattern)
+      end
+    end
+
+    it "should raise a ISO8601::Errors::TypeError if start time and end time are durations" do
+      expect { ISO8601::TimeInterval.new('P1Y2M10D/P1Y2M10D') }.to raise_error(ISO8601::Errors::TypeError)
```

Comment:
My first thought is this be a `UnknownPattern` exception instead of a `TypeError`.  The criteria applied to the rest of the library is as follows:  If the exception is due an unexpected `String` to be promoted to an ISO8601 object it is an `UnknownPattern`, if it is a (kind, `is_a?`) guard, then `TypeError`.